slack/web/classes/objects.py

```diff
@@ -0,0 +1,335 @@
+        return len(self.text) <= 300
+
+    @JsonValidator("confirm attribute cannot exceed 30 characters")
+    def confirm_length(self):
+        return len(self.confirm) <= 30
+
+    @JsonValidator("deny attribute cannot exceed 30 characters")
+    def deny_length(self):
+        return len(self.deny) <= 30
+
+    def get_json(self) -> dict:
+        self.validate_json()
+        return {
+            "title": PlainTextObject(text=self.title).get_json(),
+            "text": MarkdownTextObject(text=self.text).get_json(),
+            "confirm": PlainTextObject(text=self.confirm).get_json(),
+            "deny": PlainTextObject(text=self.deny).get_json(),
+        }
+
+
+class Option(JsonObject):
+    """
+    Option object used in dialogs, legacy message actions, and blocks
+
+    JSON should be retrieved with an explicit option_type - the Slack API has
+    different required formats in different situations
+    """
+
+    def __init__(self, *, label: str, value: str):
+        self.label = label
+        self.value = value
+
+    @JsonValidator("label attribute cannot exceed 75 characters")
+    def label_length(self):
+        return len(self.label) <= 75
+
+    @JsonValidator("value attribute cannot exceed 75 characters")
+    def value_length(self):
+        return len(self.value) <= 75
+
+    def get_json(self, option_type: str):
+        self.validate_json()
+        if option_type == "text":
+            return {
+                "text": PlainTextObject(text=self.label).get_json(),
+                "value": self.value,
+            }
+        else:
+            return {"label": self.label, "value": self.value}
+
+
+class SimpleOption(Option):
+    """
+    Simple constructor for Option objects that don't need separate labels and values
+    """
+
+    def __init__(self, *, label: str):
+        super().__init__(label=label, value=label)
+
+
+class OptionGroup(JsonObject):
+    """
+    JSON should be retrieved with an explicit option_type - the Slack API has
+    different required formats in different situations
+    """
+
+    def __init__(self, *, label: str, options: List[Option]):
+        self.label = label
+        self.options = options
+
+    @JsonValidator("label attribute cannot exceed 75 characters")
+    def label_length(self):
+        return len(self.label) <= 75
+
+    @JsonValidator("options attribute cannot exceed 100 elements")
+    def options_length(self):
+        return len(self.options) <= 100
+
+    def get_json(self, option_type: str) -> dict:
+        if option_type == "text":
+            return {
+                "label": PlainTextObject(text=self.label).get_json(),
+                "options": extract_json(self.options, Option, option_type),
+            }
+        else:
+            return {
+                "label": self.label,
+                "options": extract_json(self.options, Option, option_type),
+            }
+
+
+OptionTypes = (Option, OptionGroup)
+
+
+def extract_json(
```

Comment:
Not sure how you feel about this method - if you scan usages, it's mostly just removing boilerplate to allow us to be nicer about letting someone attach a dictionary directly where we would normally expect one of these objects